app/forms/questionnaire_form.py

```diff
@@ -148,23 +165,39 @@ def validate_mutually_exclusive_question(self, question):
             return False
         return True
 
-    def _get_target_total_and_currency(self, calculation, question):
+    def _get_target_total_and_currency(
+        self,
+        calculation: Calculation,
+        question: QuestionSchema,
+    ) -> tuple[Union[None, int, float], Optional[str]]:
+
+        calculation_value: Union[None, int, float]
+        currency: Optional[str]
+
         if "value" in calculation:
-            return calculation["value"], question.get("currency")
+            calculation_value = calculation["value"]
+            currency = question.get("currency")
+            return calculation_value, currency
 
         target_answer = self.schema.get_answers_by_answer_id(calculation["answer_id"])[
             0
         ]
-        return (
-            self.answer_store.get_answer(calculation["answer_id"]).value,
-            target_answer.get("currency"),
-        )
+        calculation_value = self.answer_store.get_answer(
+            calculation["answer_id"]
+        ).value  # type: ignore # expect not None
+        currency = target_answer.get("currency")
+
+        return calculation_value, currency
 
     def validate_date_range_with_period_limits_and_single_date_limits(
```

Comment:
We can use `NoReturn` on some of these to address the `Function is missing a return type annotation` errors

Reply:
Are there errors for this? Happy to add them but I didn't see the errors to address

Reply:
Unless i'm missing something, when I ran mypy against `questionnaire_form` I got the following errors which aren't being ignored:
```
app/forms/questionnaire_form.py:34: error: Function is missing a type annotation for one or more arguments
app/forms/questionnaire_form.py:192: error: Function is missing a return type annotation
app/forms/questionnaire_form.py:211: error: Function is missing a return type annotation
app/forms/questionnaire_form.py:223: error: Function is missing a return type annotation
app/forms/questionnaire_form.py:368: error: Returning Any from function declared to return "Decimal"
app/forms/questionnaire_form.py:398: error: Returning Any from function declared to return "str"
app/forms/questionnaire_form.py:493: error: Function is missing a return type annotation
```

Reply:
They are different ... I don't get those! How do you run yours?

Reply:
Happy to mark as resolved, change has been made to resolve remaining error and fix to ensure no uses of `NoReturn`.